sklearn/metrics/scorer.py

```diff
@@ -66,6 +67,34 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        est_has_classes = hasattr(estimator, "classes_")
+        scorer_has_labels = "labels" in signature(
+                                            self._score_func).parameters
+
+        if est_has_classes and scorer_has_labels:
+            classes = getattr(estimator, "classes_", None)
+            # if labels passed as kwargs, return kwargs as is
+            if 'labels' in self._kwargs:
+                if est_has_classes:
+                    if not np.array_equal(classes, self._kwargs['labels']):
+                        raise ValueError("`estimator classes=%r` is not the"
+                                         "same as scorer labels=%r" %
```

Comment:
And this can be valid. For instance, P/R/F allow you to exclude classes from an average, which is useful in the case of a majority "negative" class.